packages/common/src/failure.ts

```diff
@@ -53,6 +88,16 @@ export class TemporalFailure extends Error {
 
   constructor(message?: string | undefined | null, public readonly cause?: Error) {
     super(message ?? undefined);
+    this.lineage.unshift('TemporalFailure');
+  }
+
+  /**
+   * Instanceof check that is works when multiple versions of @temporalio/common are installed.
+   */
+  static is(error: unknown): error is TemporalFailure {
```

Comment:
Shouldn't we make these functions public and recommend that users also use them instead of `instanceof`? Otherwise they may be receiving errors from activity, workflow or worker and trying to handle these against a different version of common... For the same reason, I think we'll need to add `.is` checks on all error types, not only those that _we_ need to check.

Reply:
They are public.
I agree we should recommend using them instead of `instanceof`.